File: Civitas java/src/civitas/Dado.java
package civitas;
import java.util.Random;

public class Dado {
    private Random random;
    private int ultimoResultado;
    private boolean debug;
    static final private Dado instance = new Dado();
    static final private int salidaCarcel = 5;
    
    private Dado(){
        debug = false;
        ultimoResultado = 0;
        random = new Random();        
    }
    
    static public Dado getInstance(){
        return instance;
    }
    
    int tirar(){
        int genera;
        
        if(!debug){
            genera = random.nextInt(5) + 1;
        } else
            genera = 1;
        
        ultimoResultado = genera;
        return genera;
    }
    
    boolean salgoDeLaCarcel(){
        int valor = this.tirar();
        if(valor == salidaCarcel)
            return true;
        else
            return false;
    }
    
    int quienEmpieza(int n){
        return random.nextInt(n) + 1;
    }
    
    public void setDebug(boolean d){
        debug = d;
        (Diario.getInstance()).ocurreEvento("Se modifica el estado del dado debug a " + debug);
    }
    
    int getUltimoResultado(){
        return ultimoResultado;
    }
}

File: Civitas ruby/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#encoding: utf-8

#Se omite el atributo random ya que no es necesario para generar aleatorios.

require "singleton"

module Civitas
class Dado
    include Singleton
    
    @@salidaCarcel = 5
    
  def initialize
    @ultimoResultado = 0
    @debug = false
    @diario = Diario.instance
  end
  
  def tirar()
    aleatorio = 1
    
    if(!@debug)
      aleatorio = rand(1..6)
    end
    
    @ultimoResultado = aleatorio
    return aleatorio
  end
  
  def salgoDeLaCarcel
    valor = tirar
    if(valor == @@salidaCarcel)
      return true
    else
      return false
    end
  end
  
  def quienEmpieza(n)
    aleatorio = rand(1..(n-1))
    return aleatorio
  end
  
  def setDebug(d)
    @debug = d
    @diario.ocurre_evento("\nEl modo debug de dado se pone a " + @debug.to_s)
  end
  
  def getUltimoResultado
    return @ultimoResultado
  end
  
end
end
